### PR TITLE
TG Implant Refactor/Overhaul

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Completely overhauls implants to TG's standards with a little tweaking.

What is different? 

- Activation with emotes has been completely removed; activation is now handled with action buttons
 - it's hard to remember emotes in the heat of things, so this works out better. 
- There's checks to prevent implants stacking with themselves, with the exception of implants that have multiple uses. Multi-use implants will merely increase in the amount of uses with additional injections, instead (ie: +10 TC to your uplink implant, or +4 uses to freedom, etc).
- Some of the esoteric/obscure EMP features on implants have all been removed, with the exception of the Death Alarm, which behaves exactly as it currently does.
- When you perform surgery to remove an implant, if you're holding a case in your empty hand it will put the implant in the case instead of just dropping it on the table--now you can re-use imlpants. If you don't have a case, the implant will be deleted.
- Explosive implant removed
 - Microbomb implant put in its place; this acts like a hybrid of the old explosive implant and the dexplosive implant; it'll trigger on death, but you can also activate it at any time with an action button. They also stack with one another, meaning you can get bigger and bigger explosions the more you stack together.
 - Microbomb implants are only available to nuke ops; due to being similar to old C4 tactics, they won't be available to regular traitors
- Added in macrobomb implants.
 - 20 TC; just kinda the ultimate "screw you" you can possibly buy as a nuke op; causes a size 4-7-16 explosion when its triggered.
- Removes compressed matter implants
- Adds in Hidden Storage Implants instead.
 - These are similar, but better; you can activate them and open up a storage pocket only you can access that holds up to two items--want to steal the something and no one will ever be able to find it even if they search you? This is the implant for you.
- Usability improvements for things like chemical implants (self triggering, if so desired, etc).
- Loyalty Implants don't impact shadowling thralls (ie: they resist).

- Probably forgetting stuff: feel free to ask me questions!

Related Things
- Removes RNG from implant removal surgery

Things Different from TG:
- Implants are still stored in a limb as opposed to just a nebulous "contents"--this allows for injecting at specific sites, just like you're used to.

Coders
- Code isn't so shit--implant code was truly horrifying to navigate as everything was in one file, relatively pathed, and in completely illogical places.
 - Removed unused+non-funtioning deadman implant.

Maintainers (this is oddly self-referential..)
 - Forgot to mapmerge Z-2 oh well,not too worried >.>

Tested this as thoroughly as I can for any quirks and runtimes; everything seems to behave as it should and doesn't runtime, etc.

I didn't mapmerge this and forgot...oh well, it's just getting rid of two old syndicate kits that were unused and it's z-level 2 anyway...

### CODE REVIEW

**Comment by tigercat2000** [comment]
I'm concerned with removing explosive implants from standard traitors with this, as nukeops (by my knowledge) already spawned with them which activates on death.

**Comment by Deanthelis** [comment]
Will additional hidden storage implants increase the total number of pockets you have, or the size of those pockets to store objects, or are you only allowed the one?

Will an emag fit in the storage compartment?

What happens to items that are in the hidden storage implants in the off chance they're removed at some point?

Why would nukies ever bother with extra explosive implants if they can use the macrobomb one?

Are mindslave implants affected at all by these changes?

**Comment by Fox-McCloud** [comment]
>I'm concerned with removing explosive implants from standard traitors with this, as nukeops (by my knowledge) already spawned with them which activates on death.

There were two types of explosive implants--dexplosive and explosive:
- Dexplosive were the ones that automatically triggered on death and under no other circumstance (well, ok, lulz RNG emp activation).
- Explosive: Bay's weird implants that were basically the same strength as a syndicate minibomb that were tied to voice activation.
- Microbombs are kinda a combination of the two; you can trigger them, yourself, at any time with the action button, or they trigger on death--the reason I didn't re-include them is that, due to how they function, it's decently easy to permanently take people out of the round with them as they gib and spawn an explosion, therefore making it a cheap option to permanently delete brains.

@Deanthelis

- Mindslave implants are the way they've always been.
- Emags would fit into the compartment
- You would only be able to have one storage implant at a time, and no, you couldn't increase the storage slots; you get two and only two storage slots.
 - This is better than compression in every way; you can repeatedly use the storage implant over and over and over; with compression you scan and item, implant yourself, activate the item andddd that's it; you can never use it again.
- If you remove the storage implant then all items inside of it are dumped out.
- when you implant a microbomb implant into someone who already has one, it makes the boom bigger---a nukie may want to have a slightly bigger boom to ensure that when he dies, it hurts more people or deletes his own equipment so the crew can't access it.
- Macrobombs are the max size and expensive as hell; they're hat you get by stacking 10 microbombs together, essentially.

**Comment by tigercat2000** [comment]
@Fox-McCloud I was about to defend traitors with explosives, then realize the reason I was thinking of (using it for walking bombs) was never possible, so uh. nevermind.

**Comment by Deanthelis** [comment]
You can still do that if you're clever with signallers, C4, and the surgery bag. It doesn't even gib, that's the funniest part.

**Comment by Fox-McCloud** [comment]
>You can still do that if you're clever with signallers, C4, and the surgery bag. It doesn't even gib, that's the funniest part.

Yeah, there's a bot more clever things to do with explosive than explosive implants

Ever since syndicate minibombs were ported, explosive implants are hardly used for anything other than self suicide (even then..).

**Comment by Deanthelis** [comment]
Can minibombs be signaller'd like C4 can?

**Comment by Fox-McCloud** [comment]
No, they're a grenade. C4 is super special when it comes to explosives.

**Comment by Deanthelis** [comment]
I'm so used to making my own explosives when I need them, or just using C4 for everything, that I really don't know anything about the other pre-packaged exploders.

I should Traitor more.

**Comment by Plotron** [comment]
Can we implant multiple chemical implants? Do chemical implants still trigger on death?
Has the handling of chemical implants been changed in the Warden's computer?
Are removed implants still too big to fit in a toolbox?

I'm going to miss [killphrases](https://www.youtube.com/watch?v=O4wH-bU-YVY).

**Comment by Fox-McCloud** [comment]
@Plotron

- No, only one
- Yes, they do trigger on death
- No, that's unchanged.
- You won't be able to handle implants like that anymore; all implants be put into cases now.

**Comment by Plotron** [comment]
@Fox-McCloud 
But what if you don't have an empty case available? What will happen to the implant when the surgery is done?

I don't like that you can only have one chemical implant. It's a limitation that rules out some chem combinations (>50 units of goodness). Is it because of the new usability (action trigger) feature?

**Comment by Fox-McCloud** [comment]
>But what if you don't have an empty case available? What will happen to the implant when the surgery is done?

Then it gets deleted.

**Comment by Plotron** [comment]
No evidence, huh.

I *do* find it odd. Aren't some implants marked as *unknown body* in the scanner? Will the surgeon know if he removes it accidentally?

I've had a loyalty implant accidentally removed a couple of times when I was in surgery.
This new reusability feature is cool, but it will have limited potential. Empty implant cases take time to obtain.

**Comment by Fox-McCloud** [comment]
>I do find it odd. Aren't some implants marked as unknown body in the scanner? Will the surgeon know if he removes it accidentally?

Yes--they show up on the scanner as unknown (unless they're loyalty or death alarm) to help cut down on meta; the surgeon will definitely know if he removes them based on them not showing up and messages.

**Comment by FalseIncarnate** [comment]
Not map-merged, redo it.

:+1:

File: code/game/objects/items/weapons/implants/implantpad.dm [hunk 1]
```diff
@@ -1,85 +1,61 @@
-//This file was auto-corrected by findeclaration.exe on 25.5.2012 20:42:32
-
 /obj/item/weapon/implantpad
```
Comment (Markolie) [inline code/game/objects/items/weapons/implants/implantpad.dm]: Are you sure this implantpad code is an actual improvement over what we currently have? I refactored the implantpad code a while back, and from the looks of it this reverts it back to an older state, which excludes things such as proper garbage collection, a remove implant verb, etc.

**Comment by ScratKnapp** [comment]
What happens to the hidden items in the item hider implant if the implant is removed via surgery?

File: code/game/objects/items/weapons/implants/implantpad.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/implants/implantpad.dm]: @Markolie 

Wasn't aware that you refactored it; I'll take a look.

**Comment by Fox-McCloud** [comment]
>What happens to the hidden items in the item hider implant if the implant is removed via surgery?

I covered that; it drops the items.

**Comment by Fox-McCloud** [comment]
re-added the better imlant-pad code; sorry @Markolie didn't realize you overhauled that; should have looked closer.

**Comment by Fox-McCloud** [comment]
fixed an errant space.

**Comment by Fox-McCloud** [comment]
removes RNG from implant removal surgery.

**Comment by TheDZD** [comment]
Removal of pointless RNG that adds no gameplay value whatsoever? :+1: +1: +1: +1: +1:

**Pull request merged**